receiver.go

```diff
@@ -47,6 +47,7 @@ type (
 		useSessions        bool
 		sessionID          *string
 		lastError          error
+		lastErrorMu        sync.RWMutex
```

Comment:
Is there a reason we can't use `clientMu` when reading/writing last error?

Also curious if you observed actual race here or this is a proactive change.

Reply:
This was changed by one of my dev team.  You can pass in the race flag and observe it. @alexpashkov can you comment on the mutex question?

Reply:
> Is there a reason we can't use `clientMu` when reading/writing last error?
> 
> Also curious if you observed actual race here or this is a proactive change.

I've checked the code and seems like `clientMu` is used for absolutely different things, so `lastErrorMu` is reasonable. There is a race, you can observe it with a `-race` flag

Reply:
Thanks everybody for the comments on this.